Cassini/app/src/main/java/org/x/cassini/MainActivity.java

```diff
@@ -2,13 +2,133 @@
 
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.ImageView;
 import android.widget.TextView;
 
+import org.w3c.dom.Text;
+
 public class MainActivity extends AppCompatActivity {
 
+    private String TAG = "MainPage";
+    private TextView allEntries, timelineView, tags, starred, stories, settings, newEntry;
+    private ImageView allEntriesIcon, timelineViewIcon, tagsIcon, starredIcon, storiesIcon, settingsIcon, goTo1, goTo2, goTo3, goTo4, goTo5, goTo6;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        Log.d(TAG, "Entered onCreate");
+
+        // initialize various components
+        initTextView();
+        initImageView();
+        findViewById(R.id.mainpage_relative_layout).requestFocus();
+
     }
+
+    private void initTextView(){
+
+        //all entries
+        allEntries = (TextView) findViewById(R.id.main_all_entries);
+        allEntries.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View v){
+                Bundle bundle = new Bundle();
+                bundle.putString("allEntries", (String)allEntries.getText());
```

Comment:
this is redundant. allEntries.getText() will only return you the text you put in the allEntries textview, which should be 'All Entries'. This particular string is not needed for allEntriesActivity to use.